src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java

```diff
@@ -2326,4 +2326,11 @@ public RenderedMonitorTemplate renderMonitorTemplate(UUID monitorId, String reso
     }
     return renderedMonitorTemplate;
   }
+
+  public ResourceDTO findResourceByTenantIdAndResourceId(String tenantId, String resourceId) {
+    Resource resource = resourceRepository.findByTenantIdAndResourceId(tenantId, resourceId)
+        .orElse(null);
+
+    return resource == null ? null : new ResourceDTO(resource, null);
```

Comment:
`Optional` has a handy [map](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#map(java.util.function.Function)) method that lets safely remap the object only when it's present:
```suggestion
    return resourceRepository.findByTenantIdAndResourceId(tenantId, resourceId)
        .map(resource -> new ResourceDTO(resource,null));
```

Reply:
This will result in an Optional and then i have to make add condition to check if Optional is empty return null otherwise return Resource Object.

Reply:
Sorry, yes, it still needs the `.orElse(null)`